drivers/serial/uart_stm32.c

```diff
@@ -1650,34 +1649,56 @@ static int uart_stm32_async_tx(const struct device *dev,
 	/* Enable TC interrupt so we can signal correct TX done */
 	LL_USART_EnableIT_TC(usart);
 
-	/* set source address */
-	data->dma_tx.blk_cfg.source_address = (uint32_t)data->dma_tx.buffer;
-	data->dma_tx.blk_cfg.block_size = data->dma_tx.buffer_length;
+	if (!IS_ENABLED(CONFIG_UART_STM32U5_ERRATA_DMAT_LOWPOWER) ||
```

Comment:
Nit: small comment to explain this condition
```suggestion
        /**
         * Setup DMA descriptor for TX.
         * If DMAT low-power errata workaround is enabled,
         * we send the first byte using polling and the rest
         * using DMA; as such, single-byte transfers use only
         * polling and don't need to prepare a DMA descriptor.
         * In other configurations, the DMA is always used.
         */
	if (!IS_ENABLED(CONFIG_UART_STM32U5_ERRATA_DMAT_LOWPOWER) ||
```